src/main.rs

```diff
@@ -144,32 +145,48 @@ fn launch_git_with_self_as_editor() {
     run_cmd(Command::new("git").arg("commit").env("GIT_EDITOR", self_path))
 }
 
-fn collect_information_and_write_to_file(out_path: PathBuf) {
-    let maybe_emoji = select_emoji();
-
-    if maybe_emoji == None {
-        abort();
+fn git_message_is_empty(git_commit_contents: & String) -> bool {
+    for line in git_commit_contents.split("\n") {
```

Comment:
Passing a `String` (`"\n"`) to `split` creates a `Pattern`, which is unnecessary when there is only a single byte we are splitting on. Instead we want to pass a `char` (`'\n'`).

Further, I think it's better to use `.lines()` which handles Windows line endings as well:

```suggestion
    for line in git_commit_contents.lines() {
```